include/wil/resource.h

```diff
@@ -4061,6 +4023,120 @@ typedef weak_any<shared_hfind_change> weak_hfind_change;
 
 #endif // __WIL_WINBASE_STL
 
+#if (defined(_HEAPAPI_H_) && !defined(__WIL__WIL_HEAP_API) && WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM | WINAPI_PARTITION_GAMES) && !defined(WIL_KERNEL_MODE)) || \
+    defined(WIL_DOXYGEN)
+/// @cond
+#define __WIL__WIL_HEAP_API
+/// @endcond
+
+/// @cond
+namespace details
+{
+    inline void __stdcall FreeProcessHeap(_Pre_opt_valid_ _Frees_ptr_opt_ void* p)
+    {
+        ::HeapFree(::GetProcessHeap(), 0, p);
+    }
+
+    struct heap_allocator
+    {
+        static _Ret_opt_bytecap_(size) void* allocate(size_t size) WI_NOEXCEPT
+        {
+            return ::HeapAlloc(::GetProcessHeap(), HEAP_ZERO_MEMORY, size);
+        }
+    };
+
+    struct process_heap_deleter

```

Comment:
This was previously not in the `details` namespace & is probably not safe to move.

Reply:
Add to details and a #if __WIL_DEPRECATED_... wrapper around the copy outside details ?

Reply:
There are many legitimate uses of the type; deprecating is not correct either.

Reply:
I put this back.